src/directives/ripple.js

```diff
@@ -80,41 +78,80 @@ const ripple = {
 }
 
-function directive (el, binding) {
-  el.setAttribute(RippleDataAttribute, isRippleEnabled(binding))
+function rippleShow (e) {
+  const value = {}
+  const element = e.currentTarget
+  value.center = element.__ripple.centered
+  if (element.__ripple.class) {
+    value.class = element.__ripple.class
+  }
+  ripple.show(e, element, value)
+}
 
-  if ('ontouchstart' in window) {
-    el.addEventListener('touchend', () => ripple.hide(el), false)
-    el.addEventListener('touchcancel', () => ripple.hide(el), false)
+function rippleHide (e) {
+  ripple.hide(e.currentTarget)
+}
+
+function updateRipple (el, binding, wasEnabled) {
+  const enabled = isRippleEnabled(binding.value)
+  if (!enabled) {
+    ripple.hide(el)
+  }
+  el.__ripple = el.__ripple || {}
+  el.__ripple.enabled = enabled
+  const value = binding.value || {}
+  if (value.center) {
+    el.__ripple.centered = true
+  }
+  if (value.class) {
+    el.__ripple.class = binding.value.class
   }
+  if (enabled && !wasEnabled) {
+    if ('ontouchstart' in window) {
+      el.addEventListener('touchend', rippleHide, false)
+      el.addEventListener('touchcancel', rippleHide, false)
+    }
+
+    el.addEventListener('mousedown', rippleShow, false)
+    el.addEventListener('mouseup', rippleHide, false)
+    el.addEventListener('mouseleave', rippleHide, false)
+    // Anchor tags can be dragged, causes other hides to fail - #1537
+    el.addEventListener('dragstart', rippleHide, false)
+  } else if (!enabled && wasEnabled) {
+    removeListeners(el)
+  }
+}
 
-  el.addEventListener('mousedown', e => ripple.show(e, el, binding), false)
-  el.addEventListener('mouseup', () => ripple.hide(el), false)
-  el.addEventListener('mouseleave', () => ripple.hide(el), false)
-  // Anchor tags can be dragged, causes other hides to fail - #1537
-  el.addEventListener('dragstart', () => ripple.hide(el), false)
+function removeListeners (el) {
+  el.removeEventListener('touchstart', rippleShow, false)
+  el.removeEventListener('mousedown', rippleShow, false)
+  el.removeEventListener('touchend', rippleHide, false)
+  el.removeEventListener('touchcancel', rippleHide, false)
+  el.removeEventListener('mouseup', rippleHide, false)
+  el.removeEventListener('mouseleave', rippleHide, false)
+  el.removeEventListener('dragstart', rippleHide, false)
+}
+
+function directive (el, binding) {
+  updateRipple(el, binding, false)
 }
 
 function unbind (el, binding) {
-  el.removeEventListener('touchstart', e => ripple.show(e, el, binding), false)
-  el.removeEventListener('mousedown', e => ripple.show(e, el, binding), false)
-  el.removeEventListener('touchend', () => ripple.hide(el), false)
-  el.removeEventListener('touchcancel', () => ripple.hide(el), false)
-  el.removeEventListener('mouseup', () => ripple.hide(el), false)
-  el.removeEventListener('mouseleave', () => ripple.hide(el), false)
-  el.removeEventListener('dragstart', () => ripple.hide(el), false)
+  delete el.__ripple
```

Comment:
Is there anyway we could make this `_ripple` as opposed to 2 underlines? Mainly for continuity between the syntax we use for other directives for the same thing.

Reply:
I'm not sure why I used two underscores, but sure.